examples/c_api/encryption.c

```diff
@@ -130,6 +130,50 @@ void write_array() {
   tiledb_ctx_free(&ctx);
 }
 
+void write_array_without_key() {
```

Comment:
I think we should call this function `write_array` and replace the old one, since we are deprecating `tiledb_array_open_with_key`. We must remove all occurrences of `tiledb_array_open_with_key` (and the corresponding C++ API) and replace with the new way of passing the key to the config.